provider/docker/docker.go

```diff
@@ -721,6 +725,45 @@ func (p *Provider) getBasicAuth(container dockerData) []string {
 	return []string{}
 }
 
+func (p *Provider) hasRequestHeaders(container dockerData) bool {
+	if _, err := getLabel(container, types.LabelFrontendRequestHeader); err != nil {
+		return false
+	}
+	return true
+}
+
+func (p *Provider) hasResponseHeaders(container dockerData) bool {
+	if _, err := getLabel(container, types.LabelFrontendResponseHeader); err != nil {
+		return false
+	}
+	return true
+}
+
+func (p *Provider) getRequestHeaders(container dockerData) map[string]string {
+	return parseCustomHeaders(container, types.LabelFrontendRequestHeader)
+}
+
+func (p *Provider) getResponseHeaders(container dockerData) map[string]string {
+	return parseCustomHeaders(container, types.LabelFrontendResponseHeader)
+}
+
+func parseCustomHeaders(container dockerData, containerType string) map[string]string {
+	h := make(map[string]string)
+	if r, err := getLabel(container, containerType); err == nil {
+		for _, v := range strings.Split(r, ",") {
+			k := strings.Split(v, ":")
+			if len(k) != 2 {
+				log.Warnf("Could not load header %v", k)
+				return h
```

Comment:
If you add a `return` you could add `log.Errorf` instead of `log.Warnf`

In the case of an invalid key/value pair, you return the current map with previous values (or an empty map if it's the first pair), this seem strange:
- either you return a empty map (-> `log.Errorf`)
- or you only skip the invalid key/value pair (-> `log.Warnf`)